app/controllers/twitch_stream_updates_controller.rb

```diff
@@ -0,0 +1,40 @@
+class TwitchStreamUpdatesController < ApplicationController
+  skip_before_action :verify_authenticity_token
+
+  def show
+    if params["hub.mode"] == "denied"
+      Rails.logger.error("Twitch Webhook was denied: #{params.to_json}")
+      head :no_content
+    else
+      render plain: params["hub.challenge"]
+    end
+  end
+
+  def create
+    head :no_content
+
+    unless secret_verified?
+      Rails.logger.warn("Twitch Webhook Recieved for which the webhook could not be verified")
+      return
+    end
+
+    user = User.find(params[:user_id])
+
+    if params[:data].first.present?
```

Comment:
We might want to do more checking here than just that the data is present.

One thing I thought of as I wrote up the PR, is that we should probably check here that the username from the webhook matches the one the user has in their profile, to guard against getting webhooks after the profile has been updated.